c10/util/C++17.h

```diff
@@ -24,6 +24,10 @@
 #error You need C++14 to compile PyTorch
 #endif
 
+#if defined(_WIN32) && (defined(min) || defined(max))
+#  error Macro clash with min and max -- define NOMINMAX when compiling your program on Windows
```

Comment:
good catch

Reply:
This line is from pybind11. Inspired by issue #47911 .